python/pyarrow/src/arrow/python/udf.cc

```diff
@@ -101,6 +125,103 @@ struct PythonTableUdfKernelInit {
   UdfWrapperCallback cb;
 };
 
+struct PythonUdfScalarAggregatorImpl : public ScalarUdfAggregator {
+  PythonUdfScalarAggregatorImpl(UdfWrapperCallback agg_cb,
+                                std::shared_ptr<OwnedRefNoGIL> agg_function,
+                                std::vector<std::shared_ptr<DataType>> input_types,
+                                std::shared_ptr<DataType> output_type)
+      : agg_cb(agg_cb), agg_function(agg_function), output_type(output_type) {
+    Py_INCREF(agg_function->obj());
```

Comment:
This increment seems redundant given you already have one [here](https://github.com/apache/arrow/pull/35514/files#diff-d269c07da082f1b2d9b8f8628effd64aff74c163e325d253c2246410e0d1c3d0R381).

Reply:
Admitted there could be some redundancy here. I created an follow up to take a closer look:

https://github.com/apache/arrow/issues/36000